### PR TITLE
Fix cAdvisorPort, 0 is a valid option

### PR DESCRIPTION
**Pull request by mtaufen**
wrt #11710, this maintains the current default if nobody provides the flag, but allows explicitly passing 0.

/cc @farcaller @vishh @liggitt @antoineco @philips 
/assign @liggitt @vishh 

```release-note
Fixes a bug with cAdvisorPort in the KubeletConfiguration that prevented setting it to 0, which is in fact a valid option, as noted in issue #11710.
```

### CODE REVIEW

**Review by vishh — APPROVED** [review]

**Comment by antoineco** [comment]
I was certain I didn't see any warning when starting kubelet with that option, is 1.6.4 affected? (can't check, on a mobile)

**Comment by mtaufen** [comment]
Flags layer on top after this defaulting step, so there's something else funny going on if you still aren't able to override with 0 via a flag.

**Comment by antoineco** [comment]
I mean, passing 0 to that flag already disables the cAdvisor service on kube 1.6.4 (which seems strange after looking at your commit).

In which situation did you hit that bug? How can you reproduce? Just curious.

**Comment by mtaufen** [comment]
You'd hit it if you were just pulling config directly out of a KubeletConfiguration object without flags; e.g. dynamic config. I didn't personally encounter this, but it's obviously a bug so it makes sense to fix it.

**Comment by mtaufen** [comment]
It seemed like people in #11710 were having trouble with the flag though.

**Comment by antoineco** [comment]
Ah right, makes sense, good that you spotted it :clap:

**Comment by dchen1107** [comment]
The release team bug triage notes: The pr was approved after the code freeze because it indeed fixed a regression issue in 1.7 release based on the discuss at https://github.com/kubernetes/kubernetes/issues/11710#issuecomment-304630502 

cc/ @marun

**Comment by mtaufen** [comment]
re-adding label

**Comment by mtaufen** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this

**Pull request merged**
